WallsAndHoles/editor.cpp

```diff
@@ -95,17 +97,50 @@ void Editor::makeNewScene()
 
 void Editor::saveMap()
 {
-
+    if(mTileMap==nullptr){
+        QMessageBox messageBox;
+        messageBox.critical(0,"Error","TileMap doesn't exist!");
+        messageBox.setFixedSize(500,200);
+        return;
+    }
+    mTileMap->setDepend(QSharedPointer<TileTemplateSet>(mTileTemplateSet));
+    //mTileMap->updateDepend();
+    if(mTileMap->savePath().isEmpty()){
+        mTileMap->setSavePath(QFileDialog::getSaveFileName(mMainWindow,
+            tr("Save Map"), "/home", tr("Save Files (*.xml)")));
+    }
+    for(SharedTileTemplateSet set: mTileMap->dependencies()){
+        qDebug()<<"save dependencies...";
+        if(set->savePath().isEmpty()){
+            set->setSavePath(QFileDialog::getSaveFileName(mMainWindow,
+                tr("Save Templates"), "/home", tr("Save Files (*.xml)")));
+        }
+    }
+    XMLTool::saveTileMap(QSharedPointer<TileMap>(mTileMap));
 }
 
 void Editor::loadMap()
 {
+    QString fileName = QFileDialog::getOpenFileName(mMainWindow,
+        tr("Open Map"), "/home", tr("Open Files (*.xml)"));
 
+    SharedTileMap tileMap = XMLTool::openTileMap(fileName);
+    if(tileMap==nullptr){
+        QMessageBox messageBox;
+        messageBox.critical(0,"Error","Fail to load TileMap!");
+        messageBox.setFixedSize(500,200);
+        return;
+    }
+    mTileMap = tileMap.data();
+    if(!mTileMap->dependencies().isEmpty())
+        mTileTemplateSet = (mTileMap->dependencies()[0]).data();
+    mTileMapToolManager->setTileMap(mTileMap);
+    mMapView->createMap(mTileMap);
```

Comment:
@timoffex Code needs to be added here for making a new Mesher with the new tileMap. Not a big deal as that architecture will change in the future. (For now map loading doesn't create a mesh, and changing the loaded map causes a crash).